docs/docs/dev/policies/service-health-probes.md

```diff
@@ -0,0 +1,147 @@
+# Service Health Probes
+
+Kuma is able to track the status of the Envoy proxy. If grpc stream with Envoy is disconnected then Kuma considers this 
+proxy as offline, but we still send the traffic regardless of that, because this status is designed to track the connection
+between `kuma-cp` and `kuma-dp`. 
+
+Also, every `inbound` in the Dataplane model has `health` section:
+
+```yaml
+type: Dataplane
+mesh: default
+name: web-01
+networking:
+  address: 127.0.0.1
+  inbound:
+    - port: 11011
+      servicePort: 11012
+      health:
+        ready: true
+      tags:
+        kuma.io/service: backend
+        kuma.io/protocol: http
+```
+
+This `health.ready` status is intended to show the status of the service itself. It is set differently depending on 
+the environment ([Kubernetes](#kubernetes-probes) or [Universal](#universal-probes)), but it's treated the same way 
+regardless of the environment:
+
+- if `health.ready` is true or `health` section is missing - Kuma considers the inbound as healthy and includes it 
+  into load balancing set.
+- if `health.ready` is false -  Kuma doesn't include the inbound into load balancing set.
+
+Also, `health.ready` is used to compute the status of the Dataplanes and Service. You can see these statuses both in Kuma GUI and Kuma CLI:
+
+- if proxy status is `Offline`, then Dataplane is `Offline`:
+- if proxy status is `Online`:
+  - if all inbounds are ready then Dataplane is `Online`
+  - if all inbounds are not ready then Dataplane is `Offline`
+  - if at least one of the inbounds is not ready then Dataplane is `Partially degraded` 
+  - if inbound is not ready then it's not included in the load-balancer set which means it doesn't receive the traffic
+  - if all inbounds which implement the same service are ready then service is `Online`
+  - if all inbounds which implement the same service are not ready then service is `Offline`
+  - if at least one of the inbounds which implement the same service is not ready then service is `Partially degraded`
```

Comment:
We probably want to describe the different levels of health.

Each dataplane has multiple inbounds each belonging to a specific service. Each inbound is probed independently.
All inbounds in a mesh that have the same `kuma.io/service` are grouped in what is called a `service`

A service has a status which can take 3 values:

- `Online` if all dataplane inbounds with this service have `health: true`
- `Offline` if all dp inbounds with this service have `health: false`
- `PartiallyDegraded` if any inbounds with this service have `health: false`

A specific dataplane will not be present in the list of endpoints sent to dataplanes trying to communicate with this service if the inbound's matching this service health is false.

There's probably better ways to right this but I feel doing:

1. What is inbound health
2. What is a service
3. What does this mean for advertising to other dataplanes using this service

Helps comprehension.